frontend/src/routes/userLand/research/+page.svelte

```diff
@@ -126,13 +159,25 @@ function downloadCSV() {
 		console.log("downloadCSV clicked but no data to download");
 		return;
 	}
+	is_downloading = true;
 	const csvConfig = mkConfig({
 		useKeysAsHeaders: true,
 		filename: `${["mondey", new Date().toISOString().replace(/T.*/, "")].concat(selected_milestone_names).concat(selected_column_names).join("-")}`,
 		quoteStrings: true,
 	});
 	const csv = generateCsv(csvConfig)(json_data);
 	download(csvConfig)(csv);
+	is_downloading = false;
+}
+
+function handleDownloadAll() {
+	if (downloadAllHandler) {
+		is_downloading = true;
+		downloadAllHandler();
+		// Reset state
+		json_data = [];
+		plot_data = {} as PlotData;
```

Comment:
now that downloading doesn't modify json_data I guess we don't need to reset this:
```suggestion
```